march_joint_inertia_controller/include/march_joint_inertia_controller/inertia_estimator.h

```diff
@@ -0,0 +1,140 @@
+ *  used in determining the length of the velocity vector and the filtered acceleration vector. It is of specific
+ *  size to calculate the alpha value. Defaulted to 12.
+ */
+class InertiaEstimator
+{
+public:
+  /**
+   * \brief Default constructor
+   */
+  InertiaEstimator(double lambda = 0.96, size_t acc_size = 12);
+
+  double getAcceleration();
+  double getJointInertia();
+  double getJointVibration();
+
+  void setLambda(double lambda);
+  void setAccSize(size_t acc_size);
+
+  /**
+   * \brief Applies the Butterworth filter over the last two samples and returns the resulting filtered value
+   */
+  void applyButter();
+
+  /**
+   * \brief Estimate the inertia using the acceleration and torque
+   */
+  void inertiaEstimate();
+  /**
+   * \brief Calculate a discrete derivative of the speed measurements
+   */
+  double discreteSpeedDerivative(const ros::Duration& period);
+  /**
+   * \brief Calculate the alpha coefficient for the inertia estimate
+   */
+  double alphaCalculation();
+  /**
+   * \brief Calculate the inertia gain for the inertia estimate
+   */
+  double gainCalculation();
+  /**
+   * \brief Calculate the correlation coefficient of the acceleration buffer
+   */
+  void correlationCalculation();
+  /**
+   * \brief Calculate the vibration based on the acceleration
+   */
+  double vibrationCalculation();
+
+  /**
+   * \brief Fill the rolling buffers with corresponding values.
+   */
+  void fillBuffers(double velocity, double effort, const ros::Duration& period);
+
+  /**
+   * \brief Calculate the initial correlation coefficient
+   */
+  void initP(unsigned int samples);
+
+  // Vector to be filled with samples of acceleration to determine the standard deviation from
+  std::list<double> standard_deviation;
+
+private:
+  double min_alpha_ = 1.4;  // You might want to be able to adjust this value from a yaml/launch file
+  double max_alpha_ = 1.9;  // You might want to be able to adjust this value from a yaml/launch file
+
+  // This is a sixth order butterworth filter with a cutoff frequency at 15Hz in Second Order Sections form
+  std::vector<std::vector<double>> sos_{
+    { 9.16782507e-09, 1.83356501e-08, 9.16782507e-09, 1.00000000e+00, -1.82520938e+00, 8.33345838e-01 },
+    { 1.00000000e+00, 2.00000000e+00, 1.00000000e+00, 1.00000000e+00, -1.86689228e+00, 8.75214548e-01 },
+    { 1.00000000e+00, 2.00000000e+00, 1.00000000e+00, 1.00000000e+00, -1.94377925e+00, 9.52444269e-01 }
+  };
+
+  // z1 and z2 vectors are containers for previous time-step values in the butterworth filter
+  // Might be neater to design a separate class for the filter application for arbitrary filter sizes, datasets and
+  // number of signals such as: https://github.com/scipy/scipy/blob/v1.5.2/scipy/signal/_sosfilt.pyx
+  std::list<double> z1a;
+  std::list<double> z2a;
+
+  std::list<double> z1t;
+  std::list<double> z2t;
```

Comment:
I am just wondering, why did you also make these lists? You initialize them once and then always access them in order. I think this would be easier and more efficient as vectors, since they store values sequentially in memory, a vector is almost always more efficient to use. Except for the case I pointed out before, where you used the lists correctly for.